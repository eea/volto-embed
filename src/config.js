import ViewIframeBlock from './Iframe/ViewIframe.jsx';
import EditIframeBlock from './Iframe/EditIframe.jsx';

export function applyConfig(config) {
  config.blocks.blocksConfig.maps = {
    ...config.blocks.blocksConfig.maps,
    view: ViewIframeBlock,
    edit: EditIframeBlock,
  };
  return config;
}
