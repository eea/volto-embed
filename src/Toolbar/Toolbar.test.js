import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import config from '@plone/volto/registry';

import { Enlarge, FigureNote, MoreInfo, Share, Sources } from '.';

const mockStore = configureStore();

jest.mock('@plone/volto/components', () => ({
  __esModule: true,
  UniversalLink: ({ children, href }) => {
    return <a href={href}>{children}</a>;
  },
}));

jest.mock('@plone/volto-slate/editor/render', () => ({
  __esModule: true,
  serializeNodes: (nodes) => {
    return nodes.map((node, index) => {
      const Tag = node.type;
      return (
        <Tag key={`node-${index}`}>
          {node.children.map((item, index) => (
            <span key={`item-${index}`}>{item.text}</span>
          ))}
        </Tag>
      );
    });
  },
  serializeNodesToText: (nodes) => {
    return nodes
      .reduce((texts, node) => {
        if (node.children) {
          node.children.forEach((item) => {
            texts.push(item.text);
          });
        }
        return texts;
      }, [])
      .join('');
  },
}));

config.blocks.blocksConfig = {
  ...config.blocks.blocksConfig,
  maps: {
    id: 'maps',
    title: 'Maps',
    group: 'media',
    extensions: {},
    variations: [],
    restricted: false,
    mostUsed: true,
    sidebarTab: 1,
    security: {
      addPermission: [],
      view: [],
    },
  },
};

test('renders toolbar components', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
    content: {
      create: {},
    },
    connected_data_parameters: {},
  });
  const component = renderer.create(
    <Provider store={store}>
      <div className="visualization-toolbar">
        <div className="left-col">
          <FigureNote
            notes={[{ type: 'p', children: [{ text: 'This is a note' }] }]}
          />
          <Sources sources={[]} />
          <MoreInfo href={'/path/to/embeded/content'} />
        </div>
        <div className="right-col">
          <Enlarge className="enlarge-embed-maps">
            <div>Some enlarged content</div>
          </Enlarge>
          <Share href="/path/to/embeded/content" />
        </div>
      </div>
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
