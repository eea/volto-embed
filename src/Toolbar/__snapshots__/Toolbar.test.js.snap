// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders toolbar components 1`] = `
<div
  className="visualization-toolbar"
>
  <div
    className="left-col"
  >
    <div
      className="popup"
    >
      <div
        className="trigger"
      >
        <div
          className="figure-note"
        >
          <button
            className="trigger-button"
          >
            Note
          </button>
        </div>
      </div>
      <div
        className="content"
      >
        <p>
          <span>
            This is a note
          </span>
        </p>
      </div>
    </div>
    <div
      className="popup"
    >
      <div
        className="trigger"
      >
        <div
          className="sources"
        >
          <button
            className="trigger-button"
          >
            Sources
          </button>
        </div>
      </div>
      <div
        className="content"
      >
        <p>
          Data provenance is not set for this visualization.
        </p>
      </div>
    </div>
    <div
      className="more-info"
    >
      <a
        href="/path/to/embeded/content"
      >
        <span>
          More info
        </span>
        <i
          class="ri-external-link-line"
        />
      </a>
    </div>
  </div>
  <div
    className="right-col"
  >
    <div
      className="enlarge"
    >
      <button
        className="trigger-button"
        onClick={[Function]}
      >
        <i
          className="ri-fullscreen-line"
        />
        Enlarge
      </button>
    </div>
    <div
      className="popup"
    >
      <div
        className="trigger"
      >
        <div
          className="share"
        >
          <button
            className="trigger-button"
          >
            <i
              class="ri-share-fill"
            />
            <span>
              Share
            </span>
          </button>
        </div>
      </div>
      <div
        className="content"
      >
        <div
          className="item"
        >
          <span
            className="label"
          >
            Copy link
          </span>
          <div
            className="control"
          >
            <div
              className="ui input share-link"
            >
              <input
                onChange={[Function]}
                type="text"
                value="/path/to/embeded/content"
              />
            </div>
            <button
              className="ui primary button copy-button copy-button"
              onClick={[Function]}
            >
              Copy
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
